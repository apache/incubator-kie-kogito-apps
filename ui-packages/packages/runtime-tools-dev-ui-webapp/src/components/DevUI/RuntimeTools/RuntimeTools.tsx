import React from 'react';
import { HttpLink } from 'apollo-link-http';
import { onError } from 'apollo-link-error';
import { ServerUnavailablePage } from '@kogito-apps/consoles-common/dist/components/pages/ServerUnavailablePage';
import { User } from '@kogito-apps/consoles-common/dist/environment/auth';
import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory';
import ApolloClient from 'apollo-client';
import DevUIRoutes from '../DevUIRoutes/DevUIRoutes';
import DevUILayout from '../DevUILayout/DevUILayout';
import ReactDOM from 'react-dom';
import { CustomLabels } from '../../../api/CustomLabels';
import { DiagramPreviewSize } from '@kogito-apps/process-details/dist/api';

interface IOwnProps {
  isProcessEnabled: boolean;
  isTracingEnabled: boolean;
  users: User[];
  dataIndexUrl: string;
  trustyServiceUrl: string;
  navigate: string;
  devUIUrl: string;
  openApiPath: string;
  availablePages: string[];
  customLabels: CustomLabels;
  omittedProcessTimelineEvents: string[];
  diagramPreviewSize?: DiagramPreviewSize;
  isStunnerEnabled: boolean;
}

const RuntimeTools: React.FC<IOwnProps> = ({
  users,
  dataIndexUrl,
  trustyServiceUrl,
  navigate,
  devUIUrl,
  openApiPath,
  isProcessEnabled,
  isTracingEnabled,
  availablePages,
  customLabels,
  omittedProcessTimelineEvents,
  diagramPreviewSize,
  isStunnerEnabled
}) => {
  const httpLink = new HttpLink({
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    uri: dataIndexUrl
  });

  const fallbackUI = onError(({ networkError }: any) => {
    if (networkError && networkError.stack === 'TypeError: Failed to fetch') {
      // eslint-disable-next-line react/no-render-return-value
      return ReactDOM.render(
        <DevUILayout
          apolloClient={client}
          users={users}
          devUIUrl={devUIUrl}
          openApiPath={openApiPath}
          isProcessEnabled={isProcessEnabled}
          isTracingEnabled={isTracingEnabled}
          availablePages={availablePages}
          customLabels={customLabels}
          omittedProcessTimelineEvents={omittedProcessTimelineEvents}
          diagramPreviewSize={diagramPreviewSize}
          isStunnerEnabled={isStunnerEnabled}
        >
          <ServerUnavailablePage
            displayName={'Runtime Dev UI'}
            reload={() => window.location.reload()}
          />
        </DevUILayout>,
        document.getElementById('envelope-app')
      );
    }
  });

  const cache = new InMemoryCache();
  const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
    cache,
    link: fallbackUI.concat(httpLink)
  });

  return (
    <DevUILayout
      apolloClient={client}
      users={users}
      devUIUrl={devUIUrl}
      openApiPath={openApiPath}
      isProcessEnabled={isProcessEnabled}
      isTracingEnabled={isTracingEnabled}
      availablePages={availablePages}
      customLabels={customLabels}
      omittedProcessTimelineEvents={omittedProcessTimelineEvents}
      diagramPreviewSize={diagramPreviewSize}
      isStunnerEnabled={isStunnerEnabled}
    >
      <DevUIRoutes
        navigate={navigate}
        trustyServiceUrl={trustyServiceUrl}
        dataIndexUrl={dataIndexUrl}
      />
    </DevUILayout>
  );
};

export default RuntimeTools;
