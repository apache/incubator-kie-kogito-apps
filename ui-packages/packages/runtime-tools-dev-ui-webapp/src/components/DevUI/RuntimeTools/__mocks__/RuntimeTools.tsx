import React from 'react';

const RuntimeTools = () => {
  return <></>;
};

export default RuntimeTools;
