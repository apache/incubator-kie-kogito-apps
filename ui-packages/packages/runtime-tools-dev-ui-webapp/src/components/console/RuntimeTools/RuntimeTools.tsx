/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { HttpLink } from 'apollo-link-http';
import { onError } from 'apollo-link-error';
import {
  getToken,
  isAuthEnabled,
  ServerUnavailablePage,
  User
} from '@kogito-apps/consoles-common';
import { setContext } from 'apollo-link-context';
import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory';
import ApolloClient from 'apollo-client';
import ConsolesRoutes from '../ConsolesRoutes/ConsolesRoutes';
import ConsolesLayout from '../ConsolesLayout/ConsolesLayout';
import ReactDOM from 'react-dom';

interface IOwnProps {
  users: User[];
  dataIndex: string;
  navigate: string;
}

const RuntimeTools: React.FC<IOwnProps> = ({ users, dataIndex, navigate }) => {
  const httpLink = new HttpLink({
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    uri: dataIndex
  });

  const fallbackUI = onError(({ networkError }: any) => {
    if (networkError && networkError.stack === 'TypeError: Failed to fetch') {
      // eslint-disable-next-line react/no-render-return-value
      return ReactDOM.render(
        <ConsolesLayout apolloClient={client} users={users}>
          <ServerUnavailablePage
            displayName={'Runtime Dev UI'}
            reload={() => window.location.reload()}
          />
        </ConsolesLayout>,
        document.getElementById('envelope-app')
      );
    }
  });

  const setGQLContext = setContext((_, { headers }) => {
    if (isAuthEnabled()) {
      const token = getToken();
      return {
        headers: {
          ...headers,
          authorization: token ? `Bearer ${token}` : ''
        }
      };
    }
  });

  const cache = new InMemoryCache();
  const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
    cache,
    link: setGQLContext.concat(fallbackUI.concat(httpLink))
  });
  console.log('RT', users);
  return (
    <ConsolesLayout apolloClient={client} users={users}>
      <ConsolesRoutes navigate={navigate} />
    </ConsolesLayout>
  );
};

export default RuntimeTools;
