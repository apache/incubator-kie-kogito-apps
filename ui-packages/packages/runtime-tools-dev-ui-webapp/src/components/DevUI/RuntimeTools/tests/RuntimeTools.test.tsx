/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import RuntimeTools from '../RuntimeTools';

jest.mock('apollo-link-http');
jest.mock('../../DevUILayout/DevUILayout');
describe('Runtime Tools tests', () => {
  it('Snapshot tests with default props', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']} keyLength={0}>
        <RuntimeTools
          users={[{ id: 'John snow', groups: ['admin'] }]}
          dataIndexUrl="http:localhost:4000"
          trustyServiceUrl="http://localhost:1336"
          navigate="JobsManagement"
          devUIUrl="http://localhost:8080"
          openApiPath="/docs/openapi.json"
          isProcessEnabled={true}
          isTracingEnabled={true}
        />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
