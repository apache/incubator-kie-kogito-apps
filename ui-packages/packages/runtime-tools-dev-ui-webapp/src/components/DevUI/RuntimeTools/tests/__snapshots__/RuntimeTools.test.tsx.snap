// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Runtime Tools tests Snapshot tests with default props 1`] = `
<RuntimeTools
  dataIndex="http:localhost:4000"
  navigate="JobsManagement"
  users={
    Array [
      Object {
        "groups": Array [
          "admin",
        ],
        "id": "John snow",
      },
    ]
  }
/>
`;
